diffsky/experimental/disk_bulge_modeling/tests/test_disk_bulge.py

```diff
@@ -0,0 +1,291 @@
+
+        # On average, star-forming knots should be younger than diffuse disks
+        # and diffuse disks should should be younger than bulges
+        bulge_median_frac_ob_stars = np.median(bulge_age_cdf[:, indx_ostar])
+        dd_median_frac_ob_stars = np.median(dd_age_cdf[:, indx_ostar])
+        knot_median_frac_ob_stars = np.median(knot_age_cdf[:, indx_ostar])
+        assert (
+            bulge_median_frac_ob_stars
+            < dd_median_frac_ob_stars
+            < knot_median_frac_ob_stars
+        )
+
+        # Bulge SFH should never exceed total SFH
+        assert np.all(bulge_sfh <= gal_sfh)
+
+        # Bulge SFH should fall below total SFH at some point in history of each galaxy
+        assert np.all(np.any(bulge_sfh < gal_sfh, axis=1))
+
+        # Bulge fraction should respect 0 < frac_bulge < 1 for every galaxy
+        assert np.all(frac_bulge_t_obs > 0)
+        assert np.all(frac_bulge_t_obs < 1)
+        # Bulge fraction should respect 0 < frac_bulge < 1 for every galaxy
+        assert np.all(frac_bulge_t_obs > 0)
+        assert np.all(frac_bulge_t_obs < 1)
+        assert np.all(frac_bulge_t_obs < 1)
+        assert np.all(frac_bulge_t_obs < 1)
+
+
+def test_decompose_sfh_singlegal_into_bulge_disk_knots_agrees_with_vmap():
+    ran_key = jran.PRNGKey(0)
+
+    n_age = 40
+    ssp_lg_age_yr = np.linspace(5, 10.25, n_age)
+    ssp_lg_age_gyr = ssp_lg_age_yr - 9.0
+
+    n_t = 100
+    t0 = 13.8
+    t_table_min = 0.01
+    gal_t_table = np.linspace(t_table_min, t0, n_t)
+
+    n_gals = 10
+
+    itest_key, ran_key = jran.split(ran_key, 2)
+
+    # make sure t_obs > t_table_min
+    t_obs_min = 0.2
+    gal_t_obs = np.random.uniform(t_obs_min, t0 - 0.05, n_gals)
+    sfh_peak_max = np.random.uniform(0, 200)
+    sfh_peak = np.random.uniform(0, sfh_peak_max, n_gals)
+    gal_sfh_u = np.random.uniform(0, 1, n_gals * n_t).reshape((n_gals, n_t))
+    gal_sfh = gal_sfh_u * sfh_peak.reshape((n_gals, 1))
+
+    LGFBURST_UPPER_BOUND = -1
+    lgfb_max = np.random.uniform(-3, LGFBURST_UPPER_BOUND)
+    lgfb_min = np.random.uniform(lgfb_max - 3, lgfb_max)
+    gal_fburst = 10 ** np.random.uniform(lgfb_min, lgfb_max, n_gals)
+
+    gal_burstshape_params = np.tile(DEFAULT_BURST_PARAMS, n_gals)
+    gal_burstshape_params = gal_burstshape_params.reshape((n_gals, 3))
+
+    gal_burst_age_weights = _burst_age_weights_from_params_vmap(
+        ssp_lg_age_yr, gal_burstshape_params[:, 1], gal_burstshape_params[:, 2],
+    )
+    age_weights_singleburst = _burst_age_weights_from_params(
+        ssp_lg_age_yr, DEFAULT_BURST_PARAMS.lgyr_peak,
+        DEFAULT_BURST_PARAMS.lgyr_max,
+    )
+
+    age_weights_burstpop = np.tile(age_weights_singleburst, n_gals)
+    age_weights_burstpop = age_weights_burstpop.reshape((n_gals, n_age))
+    assert np.allclose(gal_burst_age_weights, age_weights_burstpop, rtol=0.001)
+
+    gal_fknot = np.random.uniform(0, FKNOT_MAX, n_gals)
```

Comment:
For stochasticity within unit tests, I try to use fixed seeds rather than random ones. Using `jax.random` makes this easy, but this can be done with numpy as well. In any case, this does not need to be changed here, but something to keep in mind for future tests.